\documentclass{article}\usepackage[]{graphicx}\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\usepackage[colorlinks=true, linkcolor=blue, citecolor=blue]{hyperref}
\usepackage[backend=bibtex,style=authoryear,backref=true,hyperref=true]{biblatex}
\addbibresource{proportionality.bib}
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{fullpage}
\usepackage{setspace}
\doublespacing

\title{RNA-Seq as a Measure of Relative Abundance: oportunities afforded by a compositional analysis framework.}
\author{Dominic LaRoche \and Dean Billheimer \and Shripad Sinari \and Kurt Michels \and  Bonnie LaFleur}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}

\maketitle

\doublespacing
\section{Abstract}


\section{Introduction}

%-- Introduce the problem and motivate the research -->
The rapid rise in the use of RNA sequencing technology (RNA-seq) for scientific discovery has led to its consideration as a clinical diagnostic tool.  However, as a new technology the analytical accuracy and reproducibility of RNA-seq must be established before it can realize its full clinical utility (\cite{SEQC/MAQC-IIIConsortium2014}, \cite{VanKeuren-Jensen2014}). Recent studies evaluating RNA-seq have found generally high intra-platform and inter-platform congruence across multiple laboratories (\cite{Li2013, tHoen2013, SEQC/MAQC-IIIConsortium2014}).  Despite these promising results, there is still a need to establish reliable diagnostics, quality control metrics and improve the reproducibility of RNA-seq data.  Understanding, and capatilizing on, the relative frequency nature of RNA-Seq data provides tools for identifying batch effects, creating quality control metrics, and improving reproducibility.\\

%Accurately identifying batch effects, and differentiating these from true biological differences, will be necessary if smaller laboratories wish to utilize RNA-seq technology for clinical applications (cite).  Moreover, the reliable identification and removal of poor quality data produced by RNA-seq pipelines has the potential to dramatically improve the analytical accuracy and reproducibility of RNA-seq data, thereby improving its clinical utility.\\

%-- Brief intro to compositional data
Relative frequency measures (hereafter reffered to as \emph{compositional data} for consistency with other disciplines) are characterzied as a vector of proportions of some whole.  These proportions are necessarily positive and sum to a constant.  The positivity and summation constraint complicate the analysis of compositions.  For example, John Aitchison (\cite{Aitchison1986}) identified the difficulty of interpreting the covariance matrix of a composition that results from the depency in the data induced by the sum constraint.  As early as 1896 Karl Pearson (\cite{Pearson1896}) identified the spurious correlation problem associated with compositions.  \\

%-- Short description of compositional data and justification of treating RNA-seq as compositional-->
NGS-based RNA-Seq methods are inherently compositional because high-throughput RNA-Seq instruments have a maximum number of reads available per run.  For example, the Roche 454 GS Junior \textsuperscript{(TM)} claims approximately 100,000 reads per run for shotgun sequencing and 70,000 reads per run for amplicon sequencing.  The Illumina Mi-Seq, with shorters read lengths, is limited to 25 million reads per sequencing run.  These reads are distributed across all of the samples included in a sequencing run and, therefore, impose a total sum constraint on the data.  This constraint cascades down to each probe or tag within a sample which is, in turn, constrained by the total number of reads allocated to the sample.  Previous authors have identified the relative nature of RNA-Seq data (\cite{Robinson2007, Anders2010, Robinson2010, Law2014, Lovell2015}).  For example, Robinson and Smyth (2007) consider counts of RNA tags as relative abundances in their development of a model for estimating differential gene expression implemented in the Bioconductor package edgeR.  Similarly, Robinson and Oshlack (2010) explicitly acknowledge the mapped read constraint when developing their widely used Trimmed-Mean of M-values (TMM) normalization method for RNA-Seq data.  \\

%-- Paragraph on existing QC metrics -->
%There are several software packages devoted to quality control of RNA-seq data (\cite{Deluca2012} and RSeQC). However, none of these proposed models address a fundamental feature of RNA-Seq data. Specifically, \



%-- Proposed solution with brief outline of methodology -->
Ignoring the sum constraint can lead to unexpected results and erroneous inference (\cite{Pearson1896, Aitchison1986, Lovell2011}).  Despite the evidence that RNA-Seq data are compositional in nature, few researchers have extended the broad set of compositional data analysis theory and tools for use in RNA-Seq analysis problems.  We extend existing compositional data methodology to include statistical diagnostic tests for the identification of sample outliers and batch effects. We also show how compositional properties can be exploited to improve exploratory analyses and improve reproducibility.\\



\section{Methods}
\subsection{Compositional Data}
%-- CODA intro -->
We begin with a brief introduction to compositional data, its properties, and some established analytical methods.  Compositional data is defined as any data in which all elements are non-negative and sum to a fixed constant (\cite{Aitchison1986}). The total sum constraint is common in biological sampling.  For example, if a 1 ml sample of blood is taken this sample could be divided into several components such as plasma, red blood cells, white blood cells, and platelets.  If the amount of any 1 component were to increase some other component (or all the other components) must decrease due to the fixed volume of the sample.  \\

%An important feature of compositional data, beyond the total sum constraint, is that of \emph{scale invariance}.  Scale invariance means that the absolute size of the total sum is irrelevent and information is only contained in the relative size between and among elements.  As previously mentioned, this feature of RNA-seq data has been recognized by previous authors (\cite{Robinson2007, Robinson2010}), although the consequences of this property have not been fully realized.\\

%-- Establish notation -->
For RNA-seq data, the total sum constraint is imposed by the limited number of available reads in each sequencing platform.  Since this total differs between platforms we will refer to the total number of available reads as $\mathbb{T}$. These reads are distributed among the $D$ samples in a sequencing run such that:
\begin{equation}
\sum_{i=1}^{D} t_i = \mathbb{T}
\label{sumt}
\end{equation}
where $t_i$ represents the total reads for sample $i$.  Because of the total sum constraint, the vector $\mathbf{t}$ is completely determined by $D-1$ elements since the $D^{th}$ element of $\mathbf{t}$ can be determined from the other $d = D-1$ elements and the total $\mathbb{T}$:  
\begin{equation}
t_D = \mathbb{T} - \sum_{i=1}^{d} t_i
\label{sumConst}
\end{equation}
In \ref{sumConst}, any of the elements can be chosen for $t_D$ with the remaining elements labeled $1, ..., d$ in any order (\cite{Aitchison1986}).  \\

From equations~\ref{sumt} and~\ref{sumConst} it is clear that the $D$ samples represent a $D - 1 = d$ dimensional simplex ($S^d$). This leads to a diffculty in interpreting the traditional $D \times D$ covariance structure.  Similarly, the compositional geometry must be accounted for when measuring the distance between two compositions or finding the center of a group of compositions (\cite{Aitchison2000}).  Because of this, standard statistical methodology is not always appropriate (\cite{Aitchison1986}) and can produce misleading results (\cite{Lovell2015}).\\

To overcome these obstacles, Aitchison (\cite{Aitchison1980}) proposed working in ratios of components. We focus on the Centered Log-Ratio (CLR) which treats the parts of the composition symmetrically and provides an informative covariance structure.  The CLR transformation is defined for a $D$-part composition $\mathbf{t}$ as:
\begin{equation}
y_i  = \text{CLR}(x_i) = log \left(\frac{x_i}{g(\mathbf{x})} \right),
\label{clr}
\end{equation}
where $g(\mathbf{t})$ is the geometric mean of $\mathbf{t}$.  The $D \times D$ covariance matrix is then defined as:
\begin{equation}
\Gamma = \left[\text{cov}\left(y_i, y_j \right): i,\ j = 1, ..., D \right]
\label{gamma}
\end{equation}
Although the CLR transformation gives equal treatment to every element of $\mathbf{t}$, the resulting covariance matrix, $\Gamma$, is singular.  Therefore, care should be taken when using general multivariate methods on CLR transformed data.\\

The constraints on compositional data also affect the measures of difference between any two compositions.  Aitchison (\cite{Aitchison1992}) outlined several properties for any compositional difference metric which must be met: scale invariance, permutation invariance, perturbation invariance (similar to translation invariance for Euclidean distance), and subcompositional dominance (similar to subspace dominance of Euclidean distance).  The scale invariance requirement is ignorable if the difference metric is applied to data on the same scale. The permutation invariance is generally satisfied by existing methods (\cite{Martin-Fernandez1998}).  However, the perturbation invariance and subcompositional dominance are not generally satisfied. Fortunately, Martin-Fernandez et al. (1998) showed that applying either Euclidean distance or Mahalanobis distance metric to CLR transformed data satisfies all the requirements of a compositional distance metric. Euclidean distance on CLR transformed compositions is referred to as Aitchison distance:

$$d_A(x_i, x_j) = \left[\sum_{k=1}^D \left( log \left(\frac{x_{ik}}{g(x_i)} \right) - log \left(\frac{x_{jk}}{g(x_j)} \right) \right)^2  \right]^\frac{1}{2}$$

or 

$$d_A(x_i, x_j) = \left[\sum_{k=1}^D \left( clr(x_{ik}) - clr(x_{jk}) \right)^2  \right]^\frac{1}{2}.$$


%Up to this point we have referred to the total reads available per sequencing run, $\mathbb{T}$.  However, it is more typical to work with the aligned reads in practice.  The total aligned reads, $T$, is always a fraction of the total reads available for a sequencing run, $\mathbb{T}$.  The fraction of the total reads aligned can be a affected by multiple factors, including the choice of alignment algorithm, which we do not address here.  We assume that $T$ imposes the same constraints on the data as oultined above for $\mathbb{T}$ and will refer exclusively to $T$ hereafter.\\  

\subsection{Outlier Detection}
Problems with RNA isolation, library preparation, or sequencing may result in a low number of reads for the sample.  There is currently no objective way to evaluate sample quality based on the total number of reads attributed to a sample. We develop a method grounded in the compositional nature of RNA-Seq data for objectively identifying samples with potentially poor quality.\\

For most experimental designs we expect the number of reads in each sample, $t_i$, to be equivalent notwithstanding random variation.   Since these reads are part of a composition it is natural to view them as arising from a Multinomial distribution with equal probabilities. Since each cell has the same probability we test for outlying values using the Binomial distribution with probability $1/D$ and size $n = \text{total available reads}$.
% The sample totals can then be ordered from the lowest, $t_{(1)}$, to the highest, $t_{(D)}$.  Outliers can then be identified as those with counts less than expected for the lowest order statistic or greater than the highest order statistic.\\



\subsection{Batch Effects and Normalization}
Batch effects arising from differing labratory conditions or operator differences have been identified as a problem in high-throughput measurement systems (\cite{leek2010, chen2011}).  Identifying and controlling for batch effects is a critical step in the transition of RNA-Seq from the lab to the clinic.  Batch effects are typically identified with a hierarchical clustering method or principal components analysis (PCA) and removed through various normalization methods (\cite{Robinson2007, Anders2010, Robinson2010, Law2014, leek2014}).  \\ 

The compositional nature of RNA-Seq data has important implications for the detection of batch effects becuase of the difficulty of interpreting the covariace matrix (\cite{Aitchison1986}) and the incompatibility with standard measures of distance (\cite{Martin-Fernandez1998}).  The CLR transformation facilitates both batch effect detection and normalization.  The CLR transformed covariance matrix is suitable for exploration through PCA (\cite{Aitchison2002}) or hierarchical clustering using standard Euclidean distance (\cite{Martin-Fernandez1998}).  \\


% *Methods Rough Outline:* 
% 
% * For CLR on total reads approach    
%     + Establish notation
%     + Introduction to compositional transformations 
%     + The CLR transformation and it's properties 
%     + Use of the CLR transformed data to identify outliers consistent with properties 
% * For use of control samples for identifying batch effects 
%     + Establish notation 
%     + Typical methods for identifying batch effects (from nature methods paper on batch effects)
%         + PCA
%         + Hierarichal clustering
%     + Apply the CLR transformation to control samples
%     + Identify batch effects using compositional methods
%         + PCA on transformed data
%         + Somehow use the set of perturbations between the control samples...
%     + Use control samples to "normalize out" batch effects in other samples somehow... (probably save this for another paper)
% 
% <!-- potential problems
% What if the control samples are so different that these differences "swamp out" the batch effects but the differences between the biological samples is not so big?
\newpage
\printbibliography
\end{document}
